stl/inc/ranges

```diff
@@ -872,6 +872,133 @@ namespace ranges {
+            }
+        }
+
+        _NODISCARD constexpr auto end() const
+            noexcept(_RANGES end(_Base) && is_nothrow_move_constructible_v<iterator_t<const _Vw>>) {
+            if constexpr (random_access_range<const _Vw> && sized_range<const _Vw>) {
+                return _RANGES begin(_Base) + _RANGES size(_Base);
+            } else {
+                return common_iterator<iterator_t<const _Vw>, sentinel_t<const _Vw>>{_RANGES end(_Base)};
+            }
+        }
+
+        _NODISCARD constexpr auto size() noexcept(
+            noexcept(_RANGES size(_Base))) /* strengthened */ requires sized_range<_Vw> {
+            return _RANGES size(_Base);
+        }
+        _NODISCARD constexpr auto size() const
+            noexcept(noexcept(_RANGES size(_Base))) /* strengthened */ requires sized_range<const _Vw> {
+            return _RANGES size(_Base);
+        }
+    };
+
+    template <class _Rng>
+    common_view(_Rng &&) -> common_view<views::all_t<_Rng>>;
+
+    namespace views {
+        // VARIABLE views::common
+        template <class _Rng>
+        concept _Can_view_all = requires(_Rng&& __r) {
+            views::all(static_cast<_Rng&&>(__r));
+        };
+
+        class _Common_fn : public _Pipe::_Base<_Common_fn> {
+        private:
+            enum class _St { _None, _All, _Common };
+
+            template <class _Rng>
+            _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                if constexpr (common_range<_Rng> && _Can_view_all<_Rng>) {
+                    return {_St::_All, noexcept(views::all{_STD declval<_Rng>()})};
+                } else if constexpr (!common_range<_Rng> && copyable<iterator_t<_Rng>>) {
+                    return {_St::_Common, noexcept(common_view{_STD declval<_Rng>()})};
+                }
+
+                return {_St::_None};
+            }
+
+            template <class _Rng>
+            static constexpr _Choice_t<_St> _Choice = _Choose<_Rng>();
+
+        public:
+            // clang-format off
+            template <viewable_range _Rng>
+                requires (_Choice<_Rng>._Strategy != _St::_None)
+            _NODISCARD constexpr auto operator()(_Rng&& _Range) const noexcept(_Choice<_Rng>._No_throw) {
+                // clang-format on
+                if constexpr (_Choice<_Rng>._Strategy == _St::_All) {
+                    return views::all{_STD forward<_Rng>(_Range)};
+                } else if constexpr (_Choice<_Rng>._Strategy == _St::_Common) {
+                    return common_view{_STD forward<_Rng>(_Range)};
+                } else {
+                    static_assert(_Always_false<_Rng>, "Should be unreachable");
+                }
+            }
+        };
+
+        inline constexpr _Common_fn common;
+
+        // ALIAS TEMPLATE views::common_t
+        template <viewable_range _Rng>
+        using common_t = decltype(common(_STD declval<_Rng>()));

```

Comment:
Leftover copypasta? There's no `common_t` in the working draft.

Reply:
Now I am hungry, ate some lines of code

Reply:
Crazy merge conflict resolution or something? There are both "new" and "old" copies of `common_view` and `_Common_fn` in this PR. Looks like we need to delete from 2165 to 2291.

Reply:
Rebased cleanly